actions/update-component-version/update-upstream.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/sh
+set -xe
+
+if [ -z "$UPSTREAM_REPO" ]
+then
+  echo "Must specify UPSTREAM_REPO"
+  exit 1
+fi
+
+if [ -z "$CHART_NAME" ]
+then
+  echo "Must specify CHART_NAME"
+  exit 1
+fi
+
+if [ -z "$GITHUB_TOKEN" ]
+then
+  echo "Must specify GITHUB_TOKEN"
+  exit 1
+fi
+
+if [[ $GITHUB_REF =~ "tags" ]]
+then
+  CHART_TAG="--tag $(echo ${GITHUB_REF} | cut -d/ -f3)"
+fi
+
+if [ -z $GIT_EMAIL ]
+then
+  echo "GIT_EMAIL must be set."
+fi
+
+if [ -z $GIT_USER ]
+then
+  echo "GIT_USER must be set."
+fi
+
+# install helm
+HELM_URL=${HELM_URL:=https://storage.googleapis.com/kubernetes-helm}
+HELM_TGZ=${HELM_TGZ:=helm-v2.14.3-linux-amd64.tar.gz}
+
+mkdir -p /tmp/helm
+wget -q ${HELM_URL}/${HELM_TGZ} -O /tmp/helm/${HELM_TGZ}
+tar -C /tmp/helm -xzv -f /tmp/helm/${HELM_TGZ}
+PATH=/tmp/helm/linux-amd64/:$PATH
+helm init --client-only
+
+# build this chart to get the version
+chartpress --skip-build $CHART_TAG
+CHART_VERSION=$(yq r helm-chart/${CHART_NAME}/Chart.yaml version)
+
+git clone --depth=1 https://${GITHUB_TOKEN}@github.com/${UPSTREAM_REPO} upstream-repo
+
+# update the upstream repo
+cd upstream-repo
+
+# set up git
+git config --global user.email "$GIT_EMAIL"
+git config --global user.name "$GIT_USER"
+
+# update the chart requirements and push
+git checkout -b auto-update/${CHART_NAME}-${CHART_VERSION} master
+yq w -i charts/renku/requirements.yaml "dependencies.(name==${CHART_NAME}).version" $CHART_VERSION
```

Comment:
I'm wondering if requiring UPSTREAM_REPO variable is something we should do assuming that we have to hardcode paths like charts/renku/requirements.yaml anyway?

Reply:
oh yeah good point - I could get rid of that

Reply:
or, alternatively, make the path to the requirements file configurable

Reply:
Yes, you can try and then we'll see how complex that would become. Do you aim to end up with something others can use? To me, that's something very renku specific but of course I could be wrong.

Reply:
yes, this one is probably pretty renku-specific - the other one (chartpress) is pretty generic

Reply:
ok I simplified it a bit